arangod/Graph/BaseOptions.cpp

```diff
@@ -470,10 +484,19 @@ void BaseOptions::serializeVariables(VPackBuilder& builder) const {
 
 void BaseOptions::setCollectionToShard(
     std::unordered_map<std::string, std::string> const& in) {
+  if (in.empty()) {
+    return;
+  }
+
   _collectionToShard.clear();
   _collectionToShard.reserve(in.size());
   for (auto const& [key, value] : in) {
-    _collectionToShard.emplace(key, std::vector{value});
+    ResourceUsageAllocator<MonitoredCollectionToShardMap> alloc = {
+        _query.resourceMonitor()};
+    auto myVec = MonitoredStringVector{alloc};
+    auto myString = MonitoredString{value, alloc};
+    myVec.emplace_back(myString);
+    _collectionToShard.emplace(key, myVec);
```

Comment:
Can the values be moved here?
```suggestion
    myVec.emplace_back(myString);
    _collectionToShard.emplace(key, myVec);
```

Reply:
Yes, added. Thanks.